File: app/models/consumer.rb
class Consumer < ActiveRecord::Base
  # The consumer model is used for reporting for the 
  # captain webhooks client. It has been decided that 
  # until fraud becomes a problem, there is no need 
  # to verify the domain to a consumer.
  
  belongs_to :producer
  has_many :webhooks
  validates_presence_of :name
end

File: app/assets/javascripts/models/consumer.js
App.Consumer = DS.Model.extend({
  name:             DS.attr('string'),
  webhookCount:    DS.attr('string'),
  producer:         DS.belongsTo('App.Producer'),
  webhooks:         DS.hasMany('App.Webhook'),
  failedWebhooks:  DS.hasMany('App.Webhook'),
});
